apps/wallet-mobile/src/features/Notifications/useCases/common/transaction-received-notification.ts

```diff
@@ -0,0 +1,156 @@
+import {useAsyncStorage} from '@yoroi/common'
+import {mountAsyncStorage} from '@yoroi/common/src'
+import {App, Notifications as NotificationTypes} from '@yoroi/types'
+import * as BackgroundFetch from 'expo-background-fetch'
+import * as TaskManager from 'expo-task-manager'
+import * as React from 'react'
+import {Subject} from 'rxjs'
+
+import {YoroiWallet} from '../../../../yoroi-wallets/cardano/types'
+import {TRANSACTION_DIRECTION} from '../../../../yoroi-wallets/types/other'
+import {useWalletManager} from '../../../WalletManager/context/WalletManagerProvider'
+import {WalletManager, walletManager} from '../../../WalletManager/wallet-manager'
+import {notificationManager} from './notification-manager'
+import {generateNotificationId} from './notifications'
+
+const BACKGROUND_FETCH_TASK = 'yoroi-transaction-received-notifications-background-fetch'
+
+// Check is needed for hot reloading, as task can not be defined twice
+if (!TaskManager.isTaskDefined(BACKGROUND_FETCH_TASK)) {
+  const appStorage = mountAsyncStorage({path: '/'})
+  TaskManager.defineTask(BACKGROUND_FETCH_TASK, async () => {
+    await syncAllWallets(walletManager)
+    const notifications = await checkForNewTransactions(walletManager, appStorage)
+    notifications.forEach((notification) => notificationManager.events.push(notification))
+    const hasNewData = notifications.length > 0
+    return hasNewData ? BackgroundFetch.BackgroundFetchResult.NewData : BackgroundFetch.BackgroundFetchResult.NoData
+  })
+}
+
+const registerBackgroundFetchAsync = () => {
+  return BackgroundFetch.registerTaskAsync(BACKGROUND_FETCH_TASK, {
+    minimumInterval: 60 * 10,
+    stopOnTerminate: false,
+    startOnBoot: true,
+  })
+}
+
+const unregisterBackgroundFetchAsync = () => {
+  return BackgroundFetch.unregisterTaskAsync(BACKGROUND_FETCH_TASK)
+}
+
+const syncAllWallets = async (walletManager: WalletManager) => {
+  const ids = [...walletManager.walletMetas.keys()]
+  const promises = ids.map((id) => {
```

Comment:
process each wallet sequentially, cuz it will fire for each wallet a lot of requests, the new packages leverage the promiseAllLimit, even though it will bloat the network.